app/src/main/java/zapsolutions/zap/transactionHistory/transactionDetails/InvoiceDetailBSDFragment.java

```diff
@@ -0,0 +1,221 @@
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.bsd_invoice_detail, container);
+
+        mTransactionDescription = view.findViewById(R.id.transactionDescription);
+        mAmountLabel = view.findViewById(R.id.amountLabel);
+        mAmount = view.findViewById(R.id.amount);
+        mMemoLabel = view.findViewById(R.id.memoLabel);
+        mMemo = view.findViewById(R.id.memo);
+        mDateLabel = view.findViewById(R.id.dateLabel);
+        mDate = view.findViewById(R.id.date);
+        mExpiryLabel = view.findViewById(R.id.expiryLabel);
+        mExpiry = view.findViewById(R.id.expiry);
+        mQRCodeView = view.findViewById(R.id.requestQRCode);
+
+        ImageButton closeButton = view.findViewById(R.id.closeButton);
+        closeButton.setOnClickListener(view1 -> dismiss());
+
+        if (getArguments() != null) {
+            ByteString transactionString = (ByteString) getArguments().getSerializable(ARGS_TRANSACTION);
+            try {
+                bindInvoice(transactionString);
+            } catch (InvalidProtocolBufferException exception) {
+                ZapLog.debug(TAG, "Failed to parse channel.");
+                dismiss();
+            } catch (NullPointerException npException) {
+                ZapLog.debug(TAG, "Failed to parse channel.");
+                dismiss();
+            }
+        }
+
+        return view;
+    }
+
+
+    private void bindInvoice(ByteString transactionString) throws InvalidProtocolBufferException {
+
+        Invoice invoice = Invoice.parseFrom(transactionString);
+
+
+        String amountLabel = getActivity().getResources().getString(R.string.amount) + ":";
+        mAmountLabel.setText(amountLabel);
+        String memoLabel = getActivity().getResources().getString(R.string.memo) + ":";
+        mMemoLabel.setText(memoLabel);
+        String dateLabel = getActivity().getResources().getString(R.string.date) + ":";
+        mDateLabel.setText(dateLabel);
+        String expiryLabel = getActivity().getResources().getString(R.string.expiry) + ":";
+        mExpiryLabel.setText(expiryLabel);
+
+        DateFormat df = DateFormat.getDateInstance(DateFormat.LONG, getActivity().getResources().getConfiguration().locale);
+        String formattedDate = df.format(new Date(invoice.getCreationDate() * 1000L));
+
+        DateFormat tf = DateFormat.getTimeInstance(DateFormat.MEDIUM, getActivity().getResources().getConfiguration().locale);
+        String formattedTime = tf.format(new Date(invoice.getCreationDate() * 1000L));
+        mDate.setText(formattedDate + ", " + formattedTime);
+
+        if (invoice.getMemo().isEmpty()) {
+            mMemo.setVisibility(View.GONE);
+            mMemoLabel.setVisibility(View.GONE);
+        } else {
+            mMemo.setText(invoice.getMemo());
+        }
+
+
+        Long amt = invoice.getValue();
```

Comment:
Just a general suggestion: I would propose to always write out the name of vars and don't shorten it, e.g. `amount` or even `invoiceAmount` to make it easier for the reader.